### DISCUSSION TITLE
Unlikely to generalize

### DISCUSSION THREAD

**Discussion by TKassis** [lucidrains/iTransformer#8] [General]
We've done a lot of work using this approach in the past (encoding individual series as tokens). It works very very well for held out data that is within the training distribution but fails catastrophically in terms of generalization to slight distribution shifts. We found that conventional ViT style tokenization (treating each series as a channel and then patchyfing) gives worse performance on held out tests sets but generalizes a lot better.

**Comment by lucidrains** [comment]
@TKassis thank you Timothy!

**Reply by lucidrains** [reply]
@TKassis you mean, patchifying as in [this work](https://arxiv.org/abs/2211.14730)?

**Reply by TKassis** [reply]
Basically exactly like ViT patchifying (in 1D) where time series multivariate are treated as channels (equivalent to RGB in ViT)

**Reply by lucidrains** [reply]
@TKassis got it! yea, let's try combining your winning approach with this work and see what happens (the time tokens i was thinking of is exactly what you describe)

**Comment by evanatyourservice** [comment]
@TKassis I'd be super interested to see when and how things fail with the itransformer setup, did you share your findings anywhere, or do you plan to in the future?

**Reply by TKassis** [reply]
No plans to do so at this time unfortunately. Just internal exploration in my group.

**Reply by evanatyourservice** [reply]
What were you using to test OOD in general?

**Reply by TKassis** [reply]
Internal datasets with some distribution shift (different sites, instruments...etc)
